substrate-node/pallets/pallet-smart-contract/src/billing.rs

```diff
@@ -734,6 +760,14 @@ impl<T: Config> Pallet<T> {
                 for ctr_id in active_node_contracts {
                     let mut ctr =
                         Contracts::<T>::get(ctr_id).ok_or(Error::<T>::ContractNotExists)?;
+
+                    // Defensive check to prevent a node contract on a rented node from transitioning to 'created' status if there's an overdraft (unsettled public IP rent).
+                    if ContractPaymentState::<T>::get(ctr_id).map_or_else(|| false, |ps| ps.has_overdraft())
```

Comment:
Just a doubt...
Why checking for overdraft here and checking for reserve at line 402 ?

Reply:
Handling different cases.
Here we check for overdraft on node contracts to prevent it from transitioning to 'created' status if there's an overdraft associated with it.

In line 402, we check for any reserved balance on node contracts to prevent the rent contract and its node contracts to transitioning to deleted state till held funds are distributed to beneficiaries.

Reply:
This allows the node contract to transition to the deleted state, thereby distributing any reserved balance accordingly. Only then, rent contract would follow in next cycle.